### PR TITLE
spring-web

### PR DESCRIPTION
**Pull request by AntonZhdanov**

### CODE REVIEW

**Review by OlegStashkiv — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -21,20 +21,32 @@
 		<maven.checkstyle.plugin.configLocation>
 			https://raw.githubusercontent.com/mate-academy/style-guides/master/java/checkstyle.xml
 		</maven.checkstyle.plugin.configLocation>
+		<mapstruct.version>1.5.5.Final</mapstruct.version>
+		<lombok.mapstrucy.binding.version>0.2.0</lombok.mapstrucy.binding.version>
 	</properties>
 	<dependencies>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter</artifactId>
```
Comment (vovasalyha) [inline pom.xml]: Is this dependency required?

File: src/main/java/com/example/bookstore/BookStoreApplication.java [hunk 2]
```diff
@@ -1,13 +1,20 @@
 package com.example.bookstore;
 
+import com.example.bookstore.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
 public class BookStoreApplication {
+    private final BookService bookService;
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/BookStoreApplication.java]: Do you use BookService here?

File: src/main/java/com/example/bookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+@Entity
+@Data
+@Table(name = "books")
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String author;
+    @Column(unique = true)
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/model/Book.java]: Specify `nullable` parameter too (and in all other not null fields, so hibernate can correctly validate you schema)

File: src/main/java/com/example/bookstore/repository/BookRepository.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import com.example.bookstore.model.Book;
+import java.util.List;
+import java.util.Optional;
+
+public interface BookRepository {
+    Book save(Book book);
+
+    List<Book> findAll();
+
+    Optional<Book> findBookById(Long id);
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/repository/BookRepository.java]:
```suggestion
    Optional<Book> findById(Long id);
```

File: src/main/java/com/example/bookstore/repository/BookRepositoryImp.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+@RequiredArgsConstructor
+@Repository
+public class BookRepositoryImp implements BookRepository {
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/repository/BookRepositoryImp.java]:
```suggestion
public class BookRepositoryImpl implements BookRepository {
```

File: src/main/java/com/example/bookstore/repository/BookRepositoryImp.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            }
+            throw e;
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            return entityManager.createQuery(
+                    "SELECT b FROM Book b", Book.class).getResultList();
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/repository/BookRepositoryImp.java]:
```suggestion
                    "FROM Book", Book.class).getResultList();
```

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ. Let's improve it a bit.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**